x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -402,6 +403,9 @@ public void executePlan(
                         coordinatorPlan,
                         localListener.acquireCompute()
                     );
+
+                    exchangeSource.onCompletion(localListener.completeImmediately());
```

Comment:
This could be delayed, possibly for a second or so, to allow "normal" completion in most of the cases.